donut/modules/calendar/static/script.js

```diff
@@ -0,0 +1,388 @@
+  }
+  $('#start_date').val(begin.substring(0, 10));
+  $('#end_date').val(end.substring(0, 10));
+  $(".deleteEvent").off("click");
+  $(".deleteEvent").on('click',{eventID:data.id, tag:data.organizer.displayName}, deleteEvent);
+  
+
+
+  // Rendering fun....
+  var eleCenter = $('#extraInfo_triangle').width()/2;
+  if (event.clientX < $('#cal').width()/2)
+  {
+    $('#extraInfo_triangle').css("left", $('#cal').width()/2 - eleCenter);
+    $('#extraInfo').css("left", $('#cal').width()/2);  
+    $('#extraInfo_triangle').css({"box-shadow": "3px -3px 3px 0px rgba(0, 0, 0, 0.4)"});
+    $('#extraInfo_triangle').css({"border-color": "white white transparent transparent"});
+  }
+  else {
+    $('#extraInfo').css("left", "0px");
+    $('#extraInfo_triangle').css("left", $('#cal').width()/2 - eleCenter);
+    $('#extraInfo_triangle').css({"box-shadow": "-3px 3px 3px 0px rgba(0, 0, 0, 0.4)"});
+    $('#extraInfo_triangle').css({"border-color": "transparent transparent white white"});
+  }
+  $('#extraInfo_triangle').css("top", 
+        Math.min($('#extraInfo').height() -  $('#extraInfo_triangle').height(), event.clientY+ 5));
+  $('#extraInfo_triangle').fadeIn();
+  $('#extraInfo').fadeIn();
+}
+
+// Render the events for a month and year on our calendar. 
+function renderEvents(month, year)
+{
+  $('div').filter($('.day')).each(function(){$(this).empty();});
+  for(var i = 0; i < events[month].length; i++)
+  {
+        
+    var curEvent =  events[month][i];
+    var checked = $(':checkbox').filter(
+     function( index ) {
+        return $(this).val() === curEvent.organizer.displayName;});
+
+    var begin = curEvent.start.dateTime || curEvent.start.date;
+    var end = curEvent.end.dateTime || curEvent.end.date;
+            
+    var startDay = parseInt(begin.substring(8, 10), 10);
+    var endDay = parseInt(end.substring(8, 10), 10);
+    var startMonth = parseInt(begin.substring(5, 7), 10);
+    var endMonth = parseInt(end.substring(5, 7), 10);
+    startDay = month > startMonth - 1 ? 1 : startDay;
+    endDay = month < endMonth - 1 ? daysOfMonth[month] : endDay;
+    for(var j = startDay; j <= endDay; j ++)
+    {
+      var divNode = document.createElement('div');
+      divNode.style.backgroundColor = colorLabel[curEvent.organizer.displayName];
+      divNode.className = 'events';
+      divNode.className += ' '+curEvent.organizer.displayName;
+      divNode.className += ' rounded-circle'; 
+      divNode.setAttribute("id", curEvent.id + j);
+      divNode.className += ' ' + curEvent.id;        
+               
+      var eventName = document.createElement('Label');
+      eventName.innerHTML = curEvent.summary;
+      divNode.appendChild(eventName);
+      if (month === d.getUTCMonth())
+      {
+        document.getElementById('day ' + j).appendChild(divNode);
+        $("#"+curEvent.id+j).on('click',{data: curEvent}, showEventInfo);
+      }
+      
+      if(!checked[0].checked){
+        divNode.style.display = "none"; 
+      }
```

Comment:
I would do all of this in jQuery instead, especially attaching click handlers to the element

Reply:
It can't seem to find ```$('#day ' + j)``` (which is generated from this file)

Reply:
Ah, you can't have spaces in CSS `id` selectors, since the space is an operator in a CSS selector. Can you remove the space character from the id?

Reply:
ooo gotcha. Thanks!